proto-lens-protoc/src/Definitions.hs

```diff
@@ -164,6 +165,33 @@ fieldName = unpack . disambiguate . camelCase
                 [""] -> error "camelCase: name consists only of underscores"
                 s':ss -> T.concat $ underlines : toLower s' : map capitalize ss
 
+-- | A list of reserved keywords in Haskell2010.
+reservedIds :: Set.Set Text
```

Comment:
Renamed to reservedKeywords.

Surprisingly "forall" and "hiding" are actually valid names:

    $ ghci -XScopedTypeVariables
    GHCi, version 7.10.2: http://www.haskell.org/ghc/  :? for help
    Prelude> let (forall, hiding) = (1,2)
    Prelude> (forall, hiding)
    (1,2)

  I added a link to the Haskell2010 report where I got the initial list from, and also added "mdo", "rec" and "proc" from that wiki page.